docs/tutorials/essentials/part-8-rtk-query-advanced.md

```diff
@@ -0,0 +1,1092 @@
+
+:::info
+
+Some of the changes in this section aren't strictly necessary - they're included to demonstrate RTK Query's features and show some of the things you _can_ do, so you can see how to use these features if you need them.
+
+:::
+
+## Editing Posts
+
+We've already added a mutation endpoint to save new Post entries to the server, and used that in our `<AddNewPostForm>`. Next, we need to handle updating the `<EditPostForm>` to let us edit an existing post.
+
+### Updating the Edit Post Form
+
+As with adding posts, the first step is to define a new mutation endpoint in our API slice. This will look much like the mutation for adding a post, but the endpoint needs to include the post ID in the URL and use an HTTP `PATCH` request to indicate that it's updating some of the fields.
+
+```js title="features/api/apiSlice.js"
+export const apiSlice = createApi({
+  reducerPath: 'api',
+  baseQuery: fetchBaseQuery({ baseUrl: '/fakeApi' }),
+  tagTypes: ['Post'],
+  endpoints: builder => ({
+    getPosts: builder.query({
+      query: () => '/posts',
+      providesTags: ['Post']
+    }),
+    getPost: builder.query({
+      query: postId => `/posts/${postId}`
+    }),
+    addNewPost: builder.mutation({
+      query: initialPost => ({
+        url: '/posts',
+        method: 'POST',
+        body: initialPost
+      }),
+      invalidatesTags: ['Post']
+    }),
+    // highlight-start
+    editPost: builder.mutation({
+      query: post => ({
+        url: `posts/${post.id}`,
+        method: 'PATCH',
+        body: post
+      })
+    })
+    // highlight-end
+  })
+})
+
+export const {
+  useGetPostsQuery,
+  useGetPostQuery,
+  useAddNewPostMutation,
+  // highlight-next-line
+  useEditPostMutation
+} = apiSlice
+```
+
+Once that's added, we can update the `<EditPostForm>`. It needs to read the original `Post` entry from the store, use that to initialize the component state to edit the fields, and then send the updated changes to the server. Currently, we're reading the `Post` entry with `selectPostById`, and manually dispatching a `postUpdated` thunk for the request.
+
+We can use the same `useGetPostQuery` hook that we used in `<SinglePostPage>` to read the `Post` entry from the cache in the store, and we'll use the new `useEditPostMutation` hook to handle saving the changes.
+
+```jsx title="features/posts/EditPostForm.js"
+import React, { useState } from 'react'
+import { useHistory } from 'react-router-dom'
+
+import { Spinner } from '../../components/Spinner'
+// highlight-next-line
+import { useGetPostQuery, useEditPostMutation } from '../api/apiSlice'
+
+export const EditPostForm = ({ match }) => {
+  const { postId } = match.params
+
+  // highlight-next-line
+  const { data: post } = useGetPostQuery(postId)
+  // highlight-next-line
+  const [updatePost, { isLoading }] = useEditPostMutation()
+
+  const [title, setTitle] = useState(post.title)
```

Comment:
Not sure if relevant for the tutorial: This would cause an `undefined` error on page refresh while in edit mode

Reply:
Yeah, there was also a similar issue with `<SinglePostPage>` before the RTKQ section. Not worth messing with for now.